src/lib/utils/__tests__/defer.ts

```diff
@@ -104,8 +104,60 @@ describe('defer', () => {
 
     expect(fn).toHaveBeenCalledTimes(0);
 
-    await Promise.resolve();
+    await deferred.wait();
 
     expect(fn).toHaveBeenCalledTimes(1);
   });
+
+  it('waits for the deferred function', async () => {
+    const fn = jest.fn();
+    const deferred = defer(fn);
+
+    deferred();
+
+    await deferred.wait();
+
+    expect(fn).toHaveBeenCalledTimes(1);
+  });
```

Comment:
Is this test really necessary?
it seems redundant with https://github.com/algolia/instantsearch.js/pull/3933/files#diff-7ad0c7bb4b40181d3ab0d95a0a4b1678R4

Reply:
Yes indeed wasn't sure about it. If it breaks then all of them break. I'll remove it.

Reply:
https://github.com/algolia/instantsearch.js/pull/3933/commits/fb1a34c908aee8b5f5cc6fb8e00f960391422ad9